WcaOnRails/app/models/competition.rb

```diff
@@ -755,8 +763,14 @@ def events_to_associated_events(events)
     end
   end
 
-  def self.search(query, params: {})
-    competitions = Competition.where(showAtAll: true)
+  def self.search(query, params: {}, current_resource_owner: nil)
+    if params[:managed_by_me].present?
+      raise WcaExceptions::BadApiParameter.new("Must be logged in to specify `managed_by_me`") unless current_resource_owner
+      raise WcaExceptions::BadApiParameter.new("Missing required scope `manage_competitions`") unless current_resource_owner.doorkeeper_token.scopes.include?("manage_competitions")
```

Comment:
I also don't really know what's usually done here, this is all new for me as well. We can't easily just redirect here because we're down in some method in a model, and redirecting only makes sense when you're in a controller.

I personally am pretty pleased with the lack of plumbing these exceptions allow us to get away with. We just throw these exceptions, and they're caught in `ApiController` with a `rescue_from WcaExceptions::BadApiParameter`](https://github.com/jfly/worldcubeassociation.org/blob/a1fd8d9530f4771a521c3714b04e196be0382115/WcaOnRails/app/controllers/api/v0/api_controller.rb#L6-L9).